feat(icons): add PackShip package builder icon

CodeNKoffee opened the pull request:
I'll help you create a PR description following the required format for the Material Icon Theme repository. Below is a properly formatted PR description:

# Description

- Added new icon for PackShip npm package builder
- Added file associations for PackShip configuration files (`.packshiprc`, `packship.config.js`, `packship.config.mjs`, `packship.config.ts`, `packship.config.mts`)
- Created a simplified SVG icon that follows Material Design guidelines and matches the theme style

## Contribution Guidelines

- [x] By creating this pull request, I acknowledge that I have read the [Contribution Guidelines](https://github.com/material-extensions/vscode-material-icon-theme/blob/main/CONTRIBUTING.md) for this project
- [x] I have read the [Code Of Conduct](https://github.com/material-extensions/vscode-material-icon-theme/blob/main/CODE_OF_CONDUCT.md) and promise to abide by these rules

CodeNKoffee reviewed (commented):
removed redundant package-lock json file created by myself accidentally

lucas-labs commented:
Hi! Thanks for your PR!

All icons in this theme have at least 1px margins.

<img width="256" src="https://github.com/user-attachments/assets/44776749-68f4-475a-ba93-f86118530566"></img>

Also, we never use pure white or black (`#ffffff`, `#000000`). This is to avoid contrast and visibility issues in different vscode themes.

You can find more info in our [contribution guidelines](https://github.com/material-extensions/vscode-material-icon-theme/blob/main/CONTRIBUTING.md).

Cheers!

CodeNKoffee commented:
> Hi! Thanks for your PR!
> 
> All icons in this theme have at least 1px margins.
> 
> <img alt="" width="256" src="https://private-user-images.githubusercontent.com/12949236/426351057-44776749-68f4-475a-ba93-f86118530566.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMwMDM5MjcsIm5iZiI6MTc0MzAwMzYyNywicGF0aCI6Ii8xMjk0OTIzNi80MjYzNTEwNTctNDQ3NzY3NDktNjhmNC00NzVhLWJhOTMtZjg2MTE4NTMwNTY2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDE1NDAyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg0NGUxYzJlYTcxNmMwODJjY2MwZTFmNTAwMTFkMzk1MjIxNWQ2ZTE1NTljNGFjNmExNTIxZGMyZTg0YmRlNzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vGu7POQk7zOwBicdJEN93e2Ydpl6Et7sC4RaTh64_TM">
> 
> Also, we never use pure white or black (`#ffffff`, `#000000`). This is to avoid contrast and visibility issues in different vscode themes.
> 
> You can find more info in our [contribution guidelines](https://github.com/material-extensions/vscode-material-icon-theme/blob/main/CONTRIBUTING.md).
> 
> Cheers!

Hey there @lucas-labs ,

Thanks for shedding light on the parts I missed. I have gone through the contribution guidelines and have amended the following as requested + in addition to adding some extra modifications:

1. Added light and dark mode icon variants
2. Simplified recognized file extensions

CodeNKoffee reviewed (commented):
Deleted `package-lock.json` accidentally added by myself.

PKief commented:
@CodeNKoffee thanks for creating the PR. I was doing a small research about PackShip and found out that it's not heavily used by many people. At least on Github.com I could only find some files of you using this project:

- https://github.com/search?q=.packshiprc&type=code

It has 1 star on Github:

<img width="451" alt="image" src="https://github.com/user-attachments/assets/3200c3cf-ca29-4ed4-8479-55587dc774dc" />

Don't get me wrong, typically we don't have the requirement of some stars or usages. But I'm also not sure if we should support every niche project. Can you maybe prove me wrong and tell me that there're more people having some use of this Pull Request?

CodeNKoffee commented:
> @CodeNKoffee thanks for creating the PR. I was doing a small research about PackShip and found out that it's not heavily used by many people. At least on Github.com I could only find some files of you using this project:
> 
> * https://github.com/search?q=.packshiprc&type=code
> 
> It has 1 star on Github:
> 
> <img alt="image" width="451" src="https://private-user-images.githubusercontent.com/12248527/427359727-3200c3cf-ca29-4ed4-8479-55587dc774dc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMyMzIzNjYsIm5iZiI6MTc0MzIzMjA2NiwicGF0aCI6Ii8xMjI0ODUyNy80MjczNTk3MjctMzIwMGMzY2YtY2EyOS00ZWQ0LTg0NzktNTU1ODdkYzc3NGRjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI5VDA3MDc0NlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVkZmFmZmVkOTE4MGExZjAwMWQzMThlOTBiNGMyYjJiYWQxNWU3ZjFhMTBhNDhiNzNhNjQxMjZmZjQzMWY1ZGEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vobLD5PfzsbeMv4xlKHRZgM4tqNELMg-jIdQGZqrGJo">
> Don't get me wrong, typically we don't have the requirement of some stars or usages. But I'm also not sure if we should support every niche project. Can you maybe prove me wrong and tell me that there're more people having some use of this Pull Request?

Thanks for your feedback! You're right that PackShip is relatively new in the open-source world. It actually started as a commercial product with several paying customers, similar to how Remix began as paid software before going open source.

While GitHub stars and public usage might seem limited now, we're in a growth phase with several contributors working to expand adoption. I have a strong track record of delivering successful projects (you can check my LinkedIn and GitHub profiles for verification), and I believe PackShip has significant potential.

- [https://linkedin.com/in/h4temsoliman](https://linkedin.com/in/h4temsoliman)
- [https://github.com/codenkoffee](https://github.com/codenkoffee)

By accepting this PR, you'd be contributing to its growth trajectory. I'm confident that with continued development and community support, PackShip could see substantial adoption within a year.

I'm sure you will be willing to reconsider the PR given this context.

Thanks in advance [https://packship.dev](https://packship.dev)

CodeNKoffee commented:
@lucas-labs @PKief can you kindly check the updated branch for the requested changes?

Thanks in advance

PKief commented:
> @lucas-labs @PKief can you kindly check the updated branch for the requested changes?
> 
> Thanks in advance

Usually we try to avoid the usage of white and black colors and use some different colors instead:

<img width="567" alt="image" src="https://github.com/user-attachments/assets/8d92244d-7674-4ead-8250-a42467fdc081" />

https://github.com/material-extensions/vscode-material-icon-theme/blob/main/CONTRIBUTING.md#use-material-design-colors

I'm not totally happy with that white border as we typically try to keep the icons as simple as possible. If it's necessary we can keep the icon like that, but I feel that a monochrome (using a single color only) version would be more beneficial.

CodeNKoffee commented:
> > @lucas-labs @PKief can you kindly check the updated branch for the requested changes?
> > Thanks in advance
> 
> Usually we try to avoid the usage of white and black colors and use some different colors instead:
> 
> <img alt="image" width="567" src="https://private-user-images.githubusercontent.com/12248527/429587796-8d92244d-7674-4ead-8250-a42467fdc081.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwMjkxMjMsIm5iZiI6MTc0NDAyODgyMywicGF0aCI6Ii8xMjI0ODUyNy80Mjk1ODc3OTYtOGQ5MjI0NGQtNzY3NC00ZWFkLTgyNTAtYTQyNDY3ZmRjMDgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA3VDEyMjcwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxZjRkN2JmYjY2OWJhZjNkYzdkZTQ0ZDQ2ZWJjMmRiMmQxN2JiNWZkZGNhYjRhM2Q3YzgxNzVkN2VhYzc4ZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MDZ3DgZBxW5B4k7WiXjaIFPf8YJcVl7pIpM7xswq-ss">
> https://github.com/material-extensions/vscode-material-icon-theme/blob/main/CONTRIBUTING.md#use-material-design-colors
> 
> I'm not totally happy with that white border as we typically try to keep the icons as simple as possible. If it's necessary we can keep the icon like that, but I feel that a monochrome (using a single color only) version would be more beneficial.

Hey! I understand that pure black and pure white aren't permitted due to their high contrast. In my original design, the white was placed within and bordered by a vibrant purple, which made it more visually balanced. However, I've updated my logo to use a single, brighter shade of purple that aligns with both themes.

Thanks again for your help and support in addition to your review in advance.

CodeNKoffee commented:
> > > @lucas-labs @PKief can you kindly check the updated branch for the requested changes?
> > > Thanks in advance
> > 
> > 
> > Usually we try to avoid the usage of white and black colors and use some different colors instead:
> > <img alt="image" width="567" src="https://private-user-images.githubusercontent.com/12248527/429587796-8d92244d-7674-4ead-8250-a42467fdc081.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDQwMjkxMjMsIm5iZiI6MTc0NDAyODgyMywicGF0aCI6Ii8xMjI0ODUyNy80Mjk1ODc3OTYtOGQ5MjI0NGQtNzY3NC00ZWFkLTgyNTAtYTQyNDY3ZmRjMDgxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA3VDEyMjcwM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQxZjRkN2JmYjY2OWJhZjNkYzdkZTQ0ZDQ2ZWJjMmRiMmQxN2JiNWZkZGNhYjRhM2Q3YzgxNzVkN2VhYzc4ZDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.MDZ3DgZBxW5B4k7WiXjaIFPf8YJcVl7pIpM7xswq-ss">
> > https://github.com/material-extensions/vscode-material-icon-theme/blob/main/CONTRIBUTING.md#use-material-design-colors
> > I'm not totally happy with that white border as we typically try to keep the icons as simple as possible. If it's necessary we can keep the icon like that, but I feel that a monochrome (using a single color only) version would be more beneficial.
> 
> Hey! I understand that pure black and pure white aren't permitted due to their high contrast. In my original design, the white was placed within and bordered by a vibrant purple, which made it more visually balanced. However, I've updated my logo to use a single, brighter shade of purple that aligns with both themes.
> 
> Thanks again for your help and support in addition to your review in advance.

All good @PKief ? 👀 😃

PKief commented:
See the changes

- [Remove DS Store files](https://github.com/material-extensions/vscode-material-icon-theme/pull/2947/commits/fd82d8d220d835f5ad1c6aa7bcfcf076bd57b5e6)
- [Reduce file size of icon by reshaping it](https://github.com/material-extensions/vscode-material-icon-theme/pull/2947/commits/5191f37aa7a624ac4dff95e390a1356806f7b7bf)

I reduced the file size of the SVG from 12KB to 2KB by recreating the shape and adjusted the color to be from Material Design Color Palette `#9575CD`.

If it's fine for you we can merge it.

PKief commented:
@CodeNKoffee actually your project is quite cool, I checked it out today and played around with it and it's some solid work. Keep going with it :)

CodeNKoffee commented:
> @CodeNKoffee actually your project is quite cool, I checked it out today and played around with it and it's some solid work. Keep going with it :)

Thanks man, appreciate the honesty and you trying it out! I strongly believe it'll grow beyond your average bedroom project.

Thanks again, stargazer ;)

CodeNKoffee commented:
> @CodeNKoffee actually your project is quite cool, I checked it out today and played around with it and it's some solid work. Keep going with it :)

Also, do you mind sharing your awesome feedback?
